### DISCUSSION TITLE
Clicking on cells / paper doesn't "un-focus" inputs

### DISCUSSION THREAD

**Discussion by alexandernst** [clientIO/joint#2665] [Q&A] [answered]
### Introduction

I just realised that my project (and the [kitchen sink demo](https://www.jointjs.com/demos/kitchen-sink-app), probably other demos too) have a very weird ¿bug?. If I click on the `Search` input in the [kitchen sink demo](https://www.jointjs.com/demos/kitchen-sink-app) and then I click on a cell (or the paper itself), the input doesn't get "un-focused", which means that hitting "Delete" on the keyboard won't delete the highlighted cell.

I surely could write a quick patch to "un-focus" the input on a few particular events (`click`, `dblclick`, `pointerdown`, etc...), but it feels dirty and I'm not really sure if there isn't a better way for handling this.

Is there a better way?

### Steps to reproduce

1.  Go to the kitchen sink demo
2. Click the "Search" input
3. Click a random cell
4. Hit the "delete" button on the keyboard

What should happen:

The cell should get deleted.

What happens:

The "delete" event is sent to the input field. If there is a text in the field, a character will be deleted.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
Does using the [preventDefaultViewAction](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#dia.Paper.prototype.options.preventDefaultViewAction) and [preventDefaultBlankAction](https://resources.jointjs.com/docs/jointjs/v4.0/joint.html#dia.Paper.prototype.options.preventDefaultBlankAction) options help?

**Reply by alexandernst** [reply]
Indeed, that is exactly what I was looking for! Thank you 🙏

**Reply by gintarasgraciovas** [reply]
perfect! :)
